### DISCUSSION TITLE
Configure from code instead of config files

### DISCUSSION THREAD

**Discussion by RobertHajbok** [ThreeMammals/Ocelot#1892] [Q&A] [answered]
## Expected Behavior / New Feature
Is there any possibility to configure Ocelot from code, maybe in Startup.cs? If not, is it in the roadmap, are you intending to add this in the future?

## Actual Behavior / Motivation for New Feature
Routes cannot be configured from the code, only from config files.

## Steps to Reproduce the Problem
Follow any setup guide from documentation.

## Specifications

  - Version: 16.0.1
  - Platform: ASP.NET 5.0

**Comment by ArchitectCat** [comment]
We did that long time ago by providing a custom `IOptionsMonitor<FileConfiguration>` implementation that was reading and building a custom Ocelot's file configuration from database instead of a file.
Here is how Ocelot registers to handle its file based configuration updates:
https://github.com/ThreeMammals/Ocelot/blob/3ef6abd7465fc77632e4b2d5189fbbf47b457867/src/Ocelot/Middleware/OcelotMiddlewareExtensions.cs#L79

All we did is implemented our own version of it (old code but will give you a basic idea):

![image](https://user-images.githubusercontent.com/43185956/122691935-3a026300-d200-11eb-905e-c13fdb3abfac.png)

**Reply by raman-m** [reply]
Hacked solution which is very specific and it has complex design.
But I love it!

**Comment by raman-m** [comment, marked as answer]
@RobertHajbok Hi Robert!
Since version 20.0 we have new DI extension for injection of a `FileConfiguration` object being constructed manually from any storages and sources including databases.
Signature: 
```csharp
public static IConfigurationBuilder AddOcelot(this IConfigurationBuilder builder, FileConfiguration fileConfiguration) {}
```
The link is [here](https://github.com/ThreeMammals/Ocelot/blob/bb795876c493af958f1d58c852ea56e94263906d/src/Ocelot/DependencyInjection/ConfigurationBuilderExtensions.cs#L105)
Commit: https://github.com/ThreeMammals/Ocelot/commit/4d245ec7fa89b35189c8e2082238ddf1f590dcf8

Hope it helps!
